packages/subproviders/src/subproviders/eth_lightwallet_subprovider.ts

```diff
@@ -0,0 +1,73 @@
+import { assert } from '@0xproject/assert';
+import { addressUtils } from '@0xproject/utils';
+import * as lightwallet from 'eth-lightwallet';
+import EthereumTx = require('ethereumjs-tx');
+import * as ethUtil from 'ethereumjs-util';
+import * as _ from 'lodash';
+
+import { PartialTxParams, WalletSubproviderErrors } from '../types';
+
+import { BaseWalletSubprovider } from './base_wallet_subprovider';
+import { PrivateKeyWalletSubprovider } from './private_key_wallet';
+
+/*
+ * This class implements the web3-provider-engine subprovider interface and forwards
+ * requests involving user accounts and signing operations to eth-lightwallet
+ *
+ * Source: https://github.com/MetaMask/provider-engine/blob/master/subproviders/subprovider.js
+ */
+export class EthLightwalletSubprovider extends BaseWalletSubprovider {
+    private _keystore: lightwallet.keystore;
+    private _pwDerivedKey: Uint8Array;
+    constructor(keystore: lightwallet.keystore, pwDerivedKey: Uint8Array) {
+        super();
+        this._keystore = keystore;
+        this._pwDerivedKey = pwDerivedKey;
+    }
+    /**
+     * Retrieve the accounts associated with the eth-lightwallet instance.
+     * This method is implicitly called when issuing a `eth_accounts` JSON RPC request
+     * via your providerEngine instance.
+     *
+     * @return An array of accounts
+     */
+    public async getAccountsAsync(): Promise<string[]> {
+        const accounts = this._keystore.getAddresses();
+        return accounts;
+    }
+    /**
+     * Signs a transaction with the account specificed by the `from` field in txParams.
+     * If you've added this Subprovider to your app's provider, you can simply send
+     * an `eth_sendTransaction` JSON RPC request, and this method will be called auto-magically.
+     * If you are not using this via a ProviderEngine instance, you can call it directly.
+     * @param txParams Parameters of the transaction to sign
+     * @return Signed transaction hex string
+     */
+    public async signTransactionAsync(txParams: PartialTxParams): Promise<string> {
+        // Lightwallet loses the chain id information when hex encoding the transaction
+        // this results in a different signature on certain networks. PrivateKeyWallet
+        // respects this as it uses the parameters passed in
+        let privKey = this._keystore.exportPrivateKey(txParams.from, this._pwDerivedKey);
+        const privKeyWallet = new PrivateKeyWalletSubprovider(privKey);
```

Comment:
We should instantiate `PrivateKeyWalletSubprovider` in the constructor so we do this once rather then for each signing call.

Reply:
I have greater concerns with keeping the private keys in memory which seems counter intuitive to eth-lightwallets key vault design.

Reply:
Ah ok, gotcha. I guess we can leave it then.